library/general/externalip/template.yaml

```diff
@@ -0,0 +1,31 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8sexternalips
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sExternalIPs
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          properties:
+            allowedIPs:
+              type: array
+              items:
+                type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8sexternalips
+
+        violation[{"msg": msg}] {
+          input.review.kind.kind == "Service"
+          input.review.kind.group == ""
+          allowedIPs := {ip | ip := input.parameters.allowedIPs[_]}
```

Comment:
Do we want to use CIDRs here? We could use `net.cidr_contains_matches`:

https://www.openpolicyagent.org/docs/latest/policy-reference/#net

Reply:
Oh neat! I think both would be useful. Is it OK if we proceed with this as-is, and I'll follow up with a PR to add an `allowedCIDRs` parameter?

Reply:
That works!